### PR TITLE
CIV-7441 Move common code in the common module

### PR DESCRIPTION
**Pull request by sabahirfan**
### JIRA link (if applicable) ###
CIV-7441

### Change description ###

This PR is continuous work for https://github.com/hmcts/civil-service/pull/2417 and moves the clients and relevant models into that module. 

Once all the common code is moved to this new module then we will extract out into it's own repo with pipelines. etc.

Will try to create small PRs to avoid conflicts with teams working in parallel.

**This is second of few PRs coming after this.** 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by asthamalviya — APPROVED** [review]

**Review by MMNycz — APPROVED** [review]

File: commons-civil/src/main/java/uk/gov/hmcts/reform/civil/bankholidays/BankHolidays.java [hunk 1]
```diff
@@ -24,15 +24,15 @@ private Countries() {
     }
 
     @JsonIgnoreProperties(ignoreUnknown = true)
-    static class Division {
+    public static class Division {
         @JsonProperty("events")
-        List<EventDate> events;
+        public List<EventDate> events;
```
Comment (filippo-possenti-version1) [inline commons-civil/src/main/java/uk/gov/hmcts/reform/civil/bankholidays/BankHolidays.java:29]: I would double-check if this is necessary once this is merged. It shouldn't.

File: commons-civil/src/main/java/uk/gov/hmcts/reform/civil/bankholidays/BankHolidays.java [hunk 2]
```diff
@@ -24,15 +24,15 @@ private Countries() {
...
 
         @JsonIgnoreProperties(ignoreUnknown = true)
-        static class EventDate {
+        public static class EventDate {
             @JsonProperty("date")
             @JsonDeserialize(using = LocalDateDeserializer.class)
-            LocalDate date;
+            public LocalDate date;
```
Comment (filippo-possenti-version1) [inline commons-civil/src/main/java/uk/gov/hmcts/reform/civil/bankholidays/BankHolidays.java:35]: I would double-check if this is necessary once this is merged. It shouldn't.

**Review by filippo-possenti-version1 — COMMENTED** [review]
I have doubts on some visibility modifiers and on a method that has been renamed (without changing the underlying implementation, which makes it a bit misleading). Aside from that, all seems fine.

File: commons-civil/src/main/java/uk/gov/hmcts/reform/civil/bankholidays/BankHolidays.java [hunk 3]
```diff
@@ -12,7 +12,7 @@
 public class BankHolidays {
 
     @JsonProperty(Countries.ENGLAND_AND_WALES)
-    Division englandAndWales;
+    public Division englandAndWales;
```
[the remark this answers is not in the document]
Comment (sabahirfan) [reply commons-civil/src/main/java/uk/gov/hmcts/reform/civil/bankholidays/BankHolidays.java:15]: yes this is as it was being used within the same package before.

File: commons-civil/src/main/java/uk/gov/hmcts/reform/civil/validation/PostcodeValidator.java [hunk 4]
```diff
@@ -16,7 +16,7 @@ public PostcodeValidator(PostcodeLookupService postcodeLookupService) {
         this.postcodeLookupService = postcodeLookupService;
     }
 
-    public List<String> validatePostCodeForDefendant(String postcode) {
+    public List<String> validate(String postcode) {
```
[the remark this answers is not in the document]
Comment (sabahirfan) [reply commons-civil/src/main/java/uk/gov/hmcts/reform/civil/validation/PostcodeValidator.java:19]: It was misleading as it was being used for both claimant and defendant. So renamed as this is generic one.

File: commons-civil/src/main/java/uk/gov/hmcts/reform/civil/bankholidays/BankHolidays.java (hunk 1 above)
Comment (sabahirfan) [reply commons-civil/src/main/java/uk/gov/hmcts/reform/civil/bankholidays/BankHolidays.java:29]: yes this is as it was being used within the same package before.

File: commons-civil/src/main/java/uk/gov/hmcts/reform/civil/bankholidays/BankHolidays.java (hunk 2 above)
Comment (sabahirfan) [reply commons-civil/src/main/java/uk/gov/hmcts/reform/civil/bankholidays/BankHolidays.java:35]: yes this is as it was being used within the same package before.

**Pull request merged**